### ISSUE TITLE
Passing node as mixin argument

### ISSUE THREAD

**Issue by logikfabrik** [AmmyUI/AmmyUI#10]
The error message is clear: "Node is not allowed as mixin argument". That's a shame.

Is this something that'll be possible in the future? Or is there some other way to achieve the same thing?

Great tool so far!

**Comment by ionoy** [comment]
Yeah, I thought about it, but still haven't figured out how to do it correctly. There are plans to pass nodes or at least node types around.

**Comment by ionoy** [comment]
By the way, could you give me an example of potential usage? How do you see it?

**Comment by logikfabrik** [comment]
Today I have a UserControl for icons. It's just a Viewbox with a Path as it's only child. The UserControl has one DependencyProperty, which just sets the Data property of the Path.

I would like to replace this UserControl with a mixin, so I can easily insert icons wherever. But to do so I need some way to pass a Geometry/GeometryGroup node into the mixin (to alternate the actual icon). Might be that this use case is a misuse of mixins, but since mixins can return nodes, it would be great if they could take them as input as well.

**Comment by ionoy** [comment]
Can you post a code example? BTW, there soon will be an addition to mixin syntax that will make them more powerful.
